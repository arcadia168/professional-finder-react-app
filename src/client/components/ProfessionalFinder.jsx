import React, { Component } from 'react';
import { Container, Row, Column } from 'react-bootstrap';
import PropTypes from 'prop-types'

class ProfessionalFinder extends Component {
    constructor(props) {
        super(props);
    }

    render() {
        return (
            <Container data-testid="pro-finder__container">

            </Container>
        );
    };
}

ProfessionalFinder.PropTypes = {
    proFinderService: PropTypes.object,
}

export default ProfessionalFinder;
