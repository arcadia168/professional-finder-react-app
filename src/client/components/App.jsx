import React, { Component } from 'react';
import { BrowserRouter as Router, Route } from "react-router-dom";
import PropTypes from 'prop-types'
import ProfessionalFinder from './professional-finder.jsx';

class App extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    return (
      <Router>
        <Route
          exact
          path="/"
          render={() => {
            return <ProfessionalFinder
              localProValues={this.props.localProValues}
              categories={this.props.categories}
              proFinderService={this.props.proFinderService}
              store={this.props.store}
            />
          }}
        />
      </Router>
    )
  }
}

App.propTypes = {
  proFinderService: PropTypes.object,
  categories: PropTypes.array,
}

export default App;
