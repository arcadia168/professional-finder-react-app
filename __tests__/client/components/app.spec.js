import React from 'react';
import App from '../../../src/client/components/app';
import renderer from 'react-test-renderer';
import path from 'path';
import fs from 'fs';
import { mount } from 'enzyme';
import ProfessionalFinder from '../../../src/client/components/professional-finder'
import proFinderService from '../../../src/client/components/professional-finder';
import ProFinderService from '../../../src/client/service/pro-finder-service';
const mockCategoriesPathName = path.resolve(__dirname, `../../__mocks__/profession-categories-mock.json`);
const mockProfessionCategories = JSON.parse(fs.readFileSync(mockCategoriesPathName, 'utf8'));
const visibleProfessionCategoriesMock = mockProfessionCategories.filter(currentProfessionCategory => !currentProfessionCategory.hidden);
xdescribe('App', () => {
    const render = customProps => {
        const props = {
            // Default props
            categories: visibleProfessionCategoriesMock,
            ...customProps,
        }
        return mount(<App {...props}/>);
    }

    it('renders the app as expected', () => {
        const component = renderer.create(
            <App
                categories={visibleProfessionCategoriesMock}
            />,
        );
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('successfully renders the app', () => {
        const renderedApp = render();
        expect(
            renderedApp.find(ProfessionalFinder).exists(),
        ).toBeTruthy();
    });

    it('assigns the proFinderService prop to the component when passed', () => {
        const mockAxios = jest.fn();
        const newProFinderService = new ProFinderService(mockAxios);
        const renderedApp = render({proFinderService: newProFinderService});
        expect(
            renderedApp.props().proFinderService,
        ).toBe(newProFinderService);
    });

    it('Passes the proFinderService down to the child component', () => {
        const mockAxios = jest.fn();
        const newProFinderService = new ProFinderService(mockAxios);
        const renderedApp = render({proFinderService: newProFinderService});
        expect(
            renderedApp.find(ProfessionalFinder).props().proFinderService,
        ).toBe(newProFinderService);
    });

    it('Assigns an object to the prop axiosInstance when passed', () => {
        const axiosInstanceMock = jest.fn();
        const renderedApp = render({ axiosInstance: axiosInstanceMock});
        expect(renderedApp.props().axiosInstance).toEqual(axiosInstanceMock);
    });
});

