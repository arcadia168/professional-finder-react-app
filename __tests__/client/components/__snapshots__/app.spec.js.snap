// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders the app as expected 1`] = `
<div
  className="pro-finder__container container"
  data-testid="pro-finder__container"
>
  <div
    className="pro-finder__title-row row"
    data-testid="pro-finder__title-row"
  >
    <div
      className="pro-finder__title-col col"
      data-testid="pro-finder__title-col"
    >
      <h1
        className="pro-finder__title"
        data-testid="pro-finder__title"
      >
        Find a Local Professional
      </h1>
    </div>
  </div>
  <div
    className="pro-finder__search-form-row row"
    data-testid="pro-finder__search-form-row"
  >
    <div
      className="pro-finder__search-form-col col"
      data-testid="pro-finder__search-form-col"
    >
      <div
        className="search-form__container"
        data-testid="search-form__container"
      >
        <div
          className="search-form__category-dropdown dropdown"
          data-testid="search-form__category-dropdown"
          onKeyDown={[Function]}
        >
          <button
            aria-expanded={false}
            aria-haspopup={true}
            className="dropdown-toggle btn btn-primary"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Choose a category
          </button>
        </div>
        <div
          className="search-form__search-field input-group"
          data-testid="search-form__search-field"
        >
          <input
            aria-label="Enter UK Postcode here..."
            className="search-form__search-field-input form-control"
            data-testid="search-form__search-field-input"
            onChange={[Function]}
            placeholder="Enter UK Postcode here..."
          />
        </div>
        <button
          className="search-form__search-btn btn btn-primary"
          data-testid="search-form__search-btn"
          disabled={false}
          onClick={[Function]}
          type="button"
        >
          Search
        </button>
      </div>
    </div>
  </div>
  <div
    className="row"
    data-testid="pro-finder__search-results-row"
  >
    <div
      className="pro-finder__search-results-table-col col"
      data-testid="pro-finder__search-results-table-col"
    >
      <div
        className="container"
        data-testid="search-form-results__container"
      >
        <div
          className="row"
        >
          <div
            className="col"
          >
            Id
          </div>
          <div
            className="col"
          >
            Name
          </div>
          <div
            className="col"
          >
            Postcode
          </div>
          <div
            className="col"
          >
            Review rating
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="pro-finder__pagination-control-row row"
    data-testid="pro-finder__pagination-control-row"
  >
    <div
      className="pro-finder__pagingation-control-col col"
      data-testid="pro-finder__pagination-control-col"
    >
      <ul
        className="pro-finder__pagination-control pagination"
        data-testid="pro-finder__pagination-control"
      />
    </div>
  </div>
</div>
`;
